plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -123,26 +110,46 @@ func (s *SelectorSpread) CalculateSpreadPriority(pod *api.Pod, nodeNameToInfo ma
 
 	// Count similar pods by node
 	countsByNodeName := map[string]int{}
-	for _, pod := range nsPods {
-		// When we are replacing a failed pod, we often see the previous deleted version
-		// while scheduling the replacement.  Ignore the previous deleted version for spreading
-		// purposes (it can still be considered for resource restrictions etc.)
-		if pod.DeletionTimestamp != nil {
-			glog.V(2).Infof("skipping pending-deleted pod: %s/%s", pod.Namespace, pod.Name)
-			continue
-		}
-		matches := false
-		for _, selector := range selectors {
-			if selector.Matches(labels.Set(pod.ObjectMeta.Labels)) {
-				matches = true
-				break
-			}
-		}
-		if !matches {
-			continue
+	countsByNodeNameLock := sync.Mutex{}
+	if len(selectors) > 0 {
+		// For every node, compute the number of pods on that node satifying
+		// any of the selectors.
+		wg := sync.WaitGroup{}
+		wg.Add(len(nodes.Items))
+		for _, node := range nodes.Items {
```

Comment:
My feeling is that go-routines are pretty cheap, and this code is much easier and more readable.
I think that since we want a more proper fix in 1.3 timeframe, this is good enough.
@davidopp ?